lib/main.dart

```diff
@@ -66,38 +69,40 @@ class _WidgetsBindingObserver extends WidgetsBindingObserver {
-  runZonedGuarded(() async {
-    // Disabling automatic system UI adjustment, which causes system nav bar
-    // color to be reverted to black when the bottom player route is being expanded.
-    //
-    // Related to https://github.com/flutter/flutter/issues/40590
-    final WidgetsBinding binding = WidgetsFlutterBinding.ensureInitialized();
-    binding.renderView.automaticSystemUiAdjustment = false;
-    await NFPrefs.initialize();
-
-    await Firebase.initializeApp();
-    if (kDebugMode) {
-      FirebaseFunctions.instance.useFunctionsEmulator('http://localhost/', 5001);
-
-      // Force disable Crashlytics collection while doing every day development.
-      // Temporarily toggle this to true if you want to test crash reporting in your app.
-      await FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(false);
-    }
+  // Disabling automatic system UI adjustment, which causes system nav bar
+  // color to be reverted to black when the bottom player route is being expanded.
+  //
+  // Related to https://github.com/flutter/flutter/issues/40590
+  final WidgetsBinding binding = WidgetsFlutterBinding.ensureInitialized();
+  binding.renderView.automaticSystemUiAdjustment = false;
+  await NFPrefs.initialize();
+
+  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
+  if (kDebugMode) {
+    FirebaseFunctions.instance.useFunctionsEmulator('http://localhost/', 5001);
+
+    // Force disable Crashlytics collection while doing every day development.
+    // Temporarily toggle this to true if you want to test crash reporting in your app.
+    await FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(false);
+  }
 
-    Isolate.current.addErrorListener(RawReceivePort((pair) async {
-      final List<dynamic> errorAndStacktrace = pair;
-      await reportError(errorAndStacktrace.first, errorAndStacktrace.last);
-    }).sendPort);
-    FlutterError.onError = reportFlutterError;
-    WidgetsBinding.instance.addObserver(_WidgetsBindingObserver());
-
-    await DeviceInfoControl.instance.init();
-    ThemeControl.instance.init();
-    ThemeControl.instance.initSystemUi();
-    await Permissions.instance.init();
-    await ContentControl.instance.init();
-    runApp(const ProviderScope(child: App()));
-  }, reportError);
+  Isolate.current.addErrorListener(RawReceivePort((pair) async {
+    final List<dynamic> errorAndStacktrace = pair;
+    await reportError(errorAndStacktrace.first, errorAndStacktrace.last);
+  }).sendPort);
+  FlutterError.onError = reportFlutterError;
+  PlatformDispatcher.instance.onError = (error, stack) {
+    reportError(error, stack);
+    return true;
+  };
+  WidgetsBinding.instance.addObserver(_WidgetsBindingObserver());
+
+  await DeviceInfoControl.instance.init();
+  ThemeControl.instance.init();
+  ThemeControl.instance.initSystemUi();
+  await Permissions.instance.init();
+  await ContentControl.instance.init();
+  runApp(const ProviderScope(child: App()));
```

Comment:
Should we also do the ` FirebaseCrashlytics.instance.recordFlutterFatalError` parat from the [docs](https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=flutter#asynchronous-errors) ?